src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -341,21 +332,19 @@ object DepositHandler extends BagValidationExtension {
   }
 
   def checkDepositIsInDraft(id: DepositId)(implicit settings: Settings): Try[Unit] = {
-    for {
-      props <- DepositProperties(id)
-      state <- props.getState
-      _ <- if (state == DRAFT) Success(())
-           else Failure(new SwordError(UriRegistry.ERROR_METHOD_NOT_ALLOWED, s"Deposit $id is not in DRAFT state."))
-    } yield ()
+    DepositProperties(id)
+      .flatMap(_.getState).flatMap {
+      case State.DRAFT => Success(())
+      case _ => Failure(new SwordError(UriRegistry.ERROR_METHOD_NOT_ALLOWED, s"Deposit $id is not in DRAFT state."))
+    }
   }
 
-  def copyPayloadToFile(deposit: Deposit, zipFile: JFile)(implicit id: DepositId): Try[Unit] =
-    try {
-      log.debug(s"[$id] Copying payload to: $zipFile")
-      Success(copyInputStreamToFile(deposit.getInputStream, zipFile))
-    } catch {
-      case t: Throwable => Failure(new SwordError(UriRegistry.ERROR_BAD_REQUEST, t))
-    }
+  def copyPayloadToFile(deposit: Deposit, zipFile: JFile)(implicit id: DepositId): Try[Unit] = Try {
+    log.debug(s"[$id] Copying payload to: $zipFile")
+    copyInputStreamToFile(deposit.getInputStream, zipFile)
+  } recover {
```

Comment:
Now you're losing an error...

```suggestion
  } recoverWith {
```